libraries/inverse/dipoleFit/dipole_fit_data.cpp

```diff
@@ -758,10 +758,10 @@ int mne_decompose_eigen_3(double *mat,
     double *dmat = MALLOC_3(np,double);
     float  *vecp = vectors[0];
 
-    const char   *uplo  = "U";
-    const char   *compz = "V";
+    __attribute__((unused)) const char   *uplo  = "U";
```

Comment:
2 Things about this:
- If these are local values that are not used, why not simply remove them instead of declaring them unused?
- Does this compile on windows? I thought this was gcc specific syntax.

Reply:
true. I checked, gcc and clang, but msvc does not. 
Today I learned about https://en.cppreference.com/w/cpp/language/attributes/maybe_unused... 
Interesting... although ahead of us. 😄  
Regarding why not deleting.... because there are some in/out paramaters which are not used and to modify them I would need to modify also the actual function calls. I it didn't seem to me so important. 
I'll roll back. (and do it right).

Reply:
Yes, I understand not changing the in/out, I was asking about using it on the local variables. Anyway, just some suggestions: Qt has Q_UNUSED that fills the same role, or we could cast to void.

Reply:
I casted to void, I'm about to push changes.
I've also reviewed all errors with clang-tidy and it seems there was another leak and a s* load of new warnings... so I'm settling.